### PR TITLE
Remove --load-version, --node and --tag flags from kedro run

### PR DESCRIPTION
**Pull request by lrcouto**
> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.
## Description

Removes flags `--load-version`, `--node`, and `--tag` from `kedro run`, mantaining only the plural versions `--load-versions`, `--nodes`, and `--tags`

Removes also the auxiliary functions `_get_values_as_tuple()` and `_reformat_load_version()`, which were used to handle formatting of the removed flags, as well as tests related to those functions.

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [x] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [x] Added tests to cover my changes
- [ ] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

**Comment by merelcht** [comment]
FYI, one thing I noticed earlier was that in `fake_run_config` we set up the config with yaml `list` syntax: https://github.com/kedro-org/kedro/blob/develop/tests/framework/cli/test_cli.py#L429-L430 

This won't work with the new flags, because we don't accept "multiple=True" anymore. See more info in https://github.com/kedro-org/kedro/issues/1792

So you'll have to update `fake_run_config` to just take in string like `"tags": "tag1, tag2",`.

**Comment by lrcouto** [comment]
Still need to update docstrings and documentation but putting it up for review so I can get feedback on the code changes.

File: kedro/framework/cli/utils.py [hunk 1]
```diff
@@ -460,37 +438,21 @@ def _split_params(ctx, param, value):
...
+
+    lv_tuple = tuple(chain.from_iterable(value.split(",") for value in [value]))
+
+    load_versions_dict = {}
+    for load_version in lv_tuple:
+        load_version = load_version.strip()  # noqa: PLW2901
+        load_version_list = load_version.split(":", 1)
+        if len(load_version_list) != 2:  # noqa: PLR2004
+            raise KedroCliError(
+                f"Expected the form of 'load_version' to be "
```
Comment (merelcht) [inline kedro/framework/cli/utils.py:452]: This should be plural `load_versions` now right?
```suggestion
                f"Expected the form of 'load_versions' to be "
```

**Review by merelcht — COMMENTED** [review]
Left one minor comments, but other than that this looks good to me 👍 Happy to approve when the doc changes are done.

**Review by stichbury — APPROVED** [review]
LGTM from docs perspective, thank you 🙏

File: RELEASE.md [hunk 2]
```diff
@@ -35,6 +35,7 @@
    * `kedro test`
 * Added the `--addons` flag to the `kedro new` command.
 * Added the `--name` flag to the `kedro new` command.
+* Deprecated `kedro run` flags `--node`, `--tag`, and `--load-version` in favour of `--nodes`, `--tags`, and `--load-versions`.
```
Comment (merelcht) [inline RELEASE.md:38]:
```suggestion
* Removed `kedro run` flags `--node`, `--tag`, and `--load-version` in favour of `--nodes`, `--tags`, and `--load-versions`.
```

**Review by merelcht — APPROVED** [review]
LGTM! 👍

**Pull request merged**